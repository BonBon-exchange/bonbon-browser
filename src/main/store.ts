/* eslint-disable no-template-curly-in-string */
/* eslint-disable import/prefer-default-export */
import Store from 'electron-store';

const store = new Store();

export const getStore = () => store;

if (store.get('browsing.defaultWebpage') === undefined)
  store.set('browsing.defaultWebpage', 'https://www.google.com');

if (store.get('browsing.searchEngine') === undefined)
  store.set('browsing.searchEngine', 'DuckDuckGo');

if (store.get('browsing.customSearchEngine') === undefined)
  store.set(
    'browsing.customSearchEngine',
    'https://duckduckgo.com/?q=${search}'
  );

if (store.get('browsing.width') === undefined) store.set('browsing.width', 600);

if (store.get('browsing.height') === undefined)
  store.set('browsing.height', 800);

if (
  store.get('browsing.size') === undefined ||
  store.get('browsing.size') === 'fit' // fix related with issue #341 https://github.com/BonBon-exchange/bonbon-web-browser/issues/341
)
  store.set('browsing.size', 'lastResized');

if (store.get('browsing.topEdge') === undefined)
  store.set('browsing.topEdge', 'fit');

// migrate searchEngine
const searchEngineValue = store.get('browsing.searchEngine') as string;
const searchEngineMigrationObject: Record<string, string> = {
  google: 'Google',
  presearch: 'Presearch',
  qwant: 'Qwant',
  duckduckgo: 'DuckDuckGo',
  yandex: 'Yandex',
  swisscows: 'Swisscows',
  ecosia: 'Ecosia',
  startpage: 'Startpage',
};
const newSearchEngineValue =
  searchEngineMigrationObject[searchEngineValue] || searchEngineValue;
store.set('browsing.searchEngine', newSearchEngineValue);
