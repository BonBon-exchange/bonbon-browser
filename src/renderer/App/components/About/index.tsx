import { About } from './About';

export * from './About';
export default About;
