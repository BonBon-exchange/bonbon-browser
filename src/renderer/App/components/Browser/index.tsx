export * from './Browser';
export { Browser as default } from './Browser';
