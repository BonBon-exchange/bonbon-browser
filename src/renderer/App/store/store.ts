import { configureStore, Store } from '@reduxjs/toolkit';
import {
  persistStore,
  persistReducer,
  createMigrate,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
  Persistor,
} from 'redux-persist';
import storage from 'redux-persist/lib/storage';

import boardReducer from './reducers/Board';
import downloadsReducer from './reducers/Downloads';
import { migrations } from './migrations';

export const store: Store = configureStore({
  reducer: {
    board: boardReducer,
    downloads: downloadsReducer,
  },
  devTools: process.env.NODE_ENV !== 'production',
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export const getPersistedStoreAndPersistor = (
  id: string
): { persistor: Persistor; store: Store } => {
  const persistConfig = {
    key: id,
    version: 3,
    storage,
    migrate: createMigrate(migrations, { debug: true }),
  };
  const persistedBoard = persistReducer(persistConfig, boardReducer);
  const persistedDownloads = persistReducer(persistConfig, downloadsReducer);
  const persistedStore = configureStore({
    reducer: {
      board: persistedBoard,
      downloads: persistedDownloads,
    },
    devTools: process.env.NODE_ENV !== 'production',
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware({
        serializableCheck: {
          ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
        },
      }),
  });
  const persistor = persistStore(persistedStore);

  return { persistor, store: persistedStore };
};
