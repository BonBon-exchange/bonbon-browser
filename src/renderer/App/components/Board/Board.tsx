/* eslint-disable promise/always-return */
/* eslint-disable @typescript-eslint/ban-ts-comment */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable import/prefer-default-export */
/* eslint-disable no-use-before-define */
import clsx from 'clsx';
import { AnimatePresence } from 'framer-motion';
import React, { useCallback, useEffect, useState } from 'react';

import { Browser } from 'renderer/App/components/Browser';
import { BrowserProps } from 'renderer/App/components/Browser/Types';
import { useBoard } from 'renderer/App/hooks/useBoard';
import { useBrowserMethods } from 'renderer/App/hooks/useBrowserMethods';
import { useStoreHelpers } from 'renderer/App/hooks/useStoreHelpers';
import { useAppDispatch } from 'renderer/App/store/hooks';
import {
  setActiveBrowser,
  setBoardHeight,
} from 'renderer/App/store/reducers/Board';

import { BoardProps } from './Types';

import './style.scss';

export const Board: React.FC<BoardProps> = ({ isFullSize, boardId }) => {
  const board = useBoard();
  const dispatch = useAppDispatch();
  const { focus } = useBrowserMethods();
  const [items, setItems] = useState<BrowserProps[]>([]);
  const helpers = useStoreHelpers();

  const boardContainer = document.querySelector('#Board__container');

  const makeBrowsers = useCallback((sorted: BrowserProps[]) => {
    return sorted.map((b) => <Browser {...b} key={b.id} firstRendering />);
  }, []);

  const contextMenuListener = (e: MouseEvent) => {
    e.preventDefault();
    window.app.tools.showBoardContextMenu({ x: e.clientX, y: e.clientY });
  };

  useEffect(() => {
    if (!board.isFullSize) {
      setTimeout(() => {
        helpers.board
          .distributeWindowsByOrder(board.browsers)
          .then(() => {
            setTimeout(
              () => board.activeBrowser && focus(board.activeBrowser),
              300
            );
          })
          .catch(console.log);
      }, 300);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [board.isFullSize]);

  useEffect(() => {
    const toSort = [...board.browsers];
    const sorted = toSort.sort((a, b) => {
      return a.id > b.id ? 1 : -1;
    });
    setItems(sorted);

    if (boardContainer) {
      const max = board.browsers.reduce(
        (acc, val) => Math.max(acc, val.top + val.height + 100),
        0
      );

      // @ts-ignore
      boardContainer.style.height = `${Number(max)}px`;

      dispatch(setBoardHeight(max));
    }
  }, [board.browsers, dispatch, boardContainer]);

  useEffect(() => {
    window.app.board.setWindowsCount({
      boardId: board.id,
      count: board.browsers.length,
    });
  }, [board.browsers.length, board.id]);

  useEffect(() => {
    if (board.activeBrowser) {
      focus(board.activeBrowser, true);
      dispatch(setActiveBrowser(board.activeBrowser));
    }
  }, [board.activeBrowser, focus, dispatch]);

  // focus activeBrowser when componentDidMount
  useEffect(() => {
    setTimeout(() => {
      if (board.activeBrowser) focus(board.activeBrowser);
      if (boardId) {
        window.app.browser
          .getUrlToOpen()
          .then((res) => {
            if (res) helpers.browser.add({ url: res });
          })
          .catch(console.log);
      }
    }, 0);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  useEffect(() => {
    document
      .getElementById('Board__container')
      ?.addEventListener('contextmenu', contextMenuListener);

    return () =>
      document
        .getElementById('Board__container')
        ?.removeEventListener('contextmenu', contextMenuListener);
  }, []);

  return (
    <div
      id="Board__container"
      className={clsx({
        'Board__is-maximized': board.isFullSize || isFullSize,
        'Board__isnt-maximized': !board.isFullSize && !isFullSize,
      })}
    >
      <AnimatePresence>{makeBrowsers(items)}</AnimatePresence>
      <div className="Board__edge-snap-left" />
      <div className="Board__edge-snap-right" />
      <div className="Board__edge-snap-maximized" />
    </div>
  );
};
