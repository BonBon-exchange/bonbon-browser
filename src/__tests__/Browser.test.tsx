/* eslint-disable react/jsx-props-no-spreading */
import '@testing-library/jest-dom';
import { render, fireEvent, screen } from '@testing-library/react';
import { Provider } from 'react-redux';

import { Browser } from '../renderer/App/components/Browser';
import { mockWindow } from './beforeAll';
import { store } from '../renderer/App/store/store';
import { toggleBoardFullSize } from '../renderer/App/store/reducers/Board';

const browserProps = {
  id: '123qsdf',
  url: 'https://www.github.com',
  top: 0,
  left: 0,
  height: 600,
  width: 800,
  firstRendering: true,
};

describe('Browser', () => {
  beforeAll(() => {
    mockWindow();
  });

  it('should render', () => {
    expect(
      render(
        <Provider store={store}>
          <Browser {...browserProps} />
        </Provider>
      )
    ).toBeTruthy();
  });

  it('should be fullsize', () => {
    store.dispatch(toggleBoardFullSize());
    const { container } = render(
      <Provider store={store}>
        <Browser {...browserProps} />
      </Provider>
    );

    const browser = container.getElementsByClassName('Browser__is-full-size');
    expect(browser.length).toBe(1);
  });

  it('should toggle fullsize', () => {
    render(
      <Provider store={store}>
        <Browser {...browserProps} />
      </Provider>
    );

    fireEvent.click(screen.getByTestId('toggle-enlarge-browser'));

    expect(
      screen
        .getByTestId('browser-window')
        .className.includes('Browser__is-full-size')
    ).toBeFalsy();
  });
});
