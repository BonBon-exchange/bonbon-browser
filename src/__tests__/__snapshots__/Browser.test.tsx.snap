// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Browser should be fullsize 1`] = `
"<div style=\\"position: absolute; user-select: auto; width: 800px; height: 600px; display: flex; top: 0px; left: 0px; cursor: auto; transform: translate(0px,0px); max-width: 9007199254740991px; max-height: 9007199254740991px; box-sizing: border-box; flex-shrink: 0; z-index: 2;\\" class=\\"Browser__is-full-size Browser__draggable-container react-draggable\\" id=\\"Browser__123qsdf\\" data-testid=\\"browser-window\\" data-id=\\"123qsdf\\">
  <div class=\\"Browser__container\\" style=\\"opacity: 0;\\">
    <div class=\\"BrowserTopBar__container\\">
      <div class=\\"BrowserTopBar__title\\"></div>
      <div class=\\"BrowserTopBar__controls\\">
        <div class=\\"BrowserTopBar__control-button close-button\\" data-testid=\\"close-browser\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"CloseIcon\\">
            <path d=\\"M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z\\"></path>
          </svg></div>
        <div class=\\"BrowserTopBar__control-button\\" data-testid=\\"toggle-enlarge-browser\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"16\\" height=\\"16\\" viewBox=\\"0 0 16 16\\" fill=\\"currentColor\\">
            <path d=\\"M3 5v9h9V5H3zm8 8H4V6h7v7z\\"></path>
            <path fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M5 5h1V4h7v7h-1v1h2V3H5v2z\\"></path>
          </svg></div>
        <div class=\\"BrowserTopBar__control-button\\" data-testid=\\"minimize-browser\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"MinimizeIcon\\">
            <path d=\\"M6 19h12v2H6z\\"></path>
          </svg></div>
      </div>
    </div>
    <div class=\\"BrowserControlBar__container\\">
      <div class=\\"BrowserControlBar__controls\\">
        <div class=\\"BrowserControlBar__control\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"ArrowBackIcon\\">
            <path d=\\"M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z\\"></path>
          </svg></div>
        <div class=\\"BrowserControlBar__control\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"ArrowForwardIcon\\">
            <path d=\\"m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z\\"></path>
          </svg></div>
        <div class=\\"BrowserControlBar__control\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"CachedIcon\\">
            <path d=\\"m19 8-4 4h3c0 3.31-2.69 6-6 6-1.01 0-1.97-.25-2.8-.7l-1.46 1.46C8.97 19.54 10.43 20 12 20c4.42 0 8-3.58 8-8h3l-4-4zM6 12c0-3.31 2.69-6 6-6 1.01 0 1.97.25 2.8.7l1.46-1.46C15.03 4.46 13.57 4 12 4c-4.42 0-8 3.58-8 8H1l4 4 4-4H6z\\"></path>
          </svg></div>
        <div class=\\"BrowserControlBar__control\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"HomeIcon\\">
            <path d=\\"M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z\\"></path>
          </svg></div>
      </div>
      <div class=\\"MuiFormControl-root MuiTextField-root BrowserControlBar_url-input css-1u3bzj6-MuiFormControl-root-MuiTextField-root\\">
        <div class=\\"MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1ptx2yq-MuiInputBase-root-MuiInput-root\\"><input aria-invalid=\\"false\\" id=\\":r0:\\" type=\\"text\\" class=\\"MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input\\" value=\\"https://www.github.com\\"></div>
      </div>
      <div class=\\"BrowserControlBar__bookmark-control\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"StarBorderIcon\\">
          <path d=\\"m22 9.24-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z\\"></path>
        </svg></div>
    </div>
    <div class=\\"Browser__webview-container\\">
      <webview allowpopups=\\"true\\" src=\\"https://www.github.com\\" partition=\\"persist:user-partition\\"></webview>
    </div>
  </div>
  <div></div>
</div>"
`;

exports[`Browser should render 1`] = `
<div
  className="Browser__draggable-container react-draggable"
  data-id="123qsdf"
  data-testid="browser-window"
  id="Browser__123qsdf"
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  style={
    Object {
      "boxSizing": "border-box",
      "cursor": "auto",
      "display": "flex",
      "flexShrink": 0,
      "height": "600px",
      "left": 0,
      "maxHeight": 9007199254740991,
      "maxWidth": 9007199254740991,
      "minHeight": undefined,
      "minWidth": undefined,
      "position": "absolute",
      "top": 0,
      "transform": "translate(0px,0px)",
      "userSelect": "auto",
      "width": "800px",
    }
  }
  transform={null}
>
  <div
    className="Browser__container"
    style={
      Object {
        "opacity": 0,
      }
    }
  >
    <div
      className="BrowserTopBar__container"
      onClick={[Function]}
      onDoubleClick={[Function]}
    >
      <div
        className="BrowserTopBar__title"
      />
      <div
        className="BrowserTopBar__controls"
      >
        <div
          className="BrowserTopBar__control-button close-button"
          data-testid="close-browser"
          onClick={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </div>
        <div
          className="BrowserTopBar__control-button"
          data-testid="toggle-enlarge-browser"
          onClick={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CropSquareIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18 4H6c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H6V6h12v12z"
            />
          </svg>
        </div>
        <div
          className="BrowserTopBar__control-button"
          data-testid="minimize-browser"
          onClick={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="MinimizeIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19h12v2H6z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="BrowserControlBar__container"
    >
      <div
        className="BrowserControlBar__controls"
      >
        <div
          className="BrowserControlBar__control"
          onClick={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowBackIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
        </div>
        <div
          className="BrowserControlBar__control"
          onClick={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowForwardIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
            />
          </svg>
        </div>
        <div
          className="BrowserControlBar__control"
          onClick={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CachedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m19 8-4 4h3c0 3.31-2.69 6-6 6-1.01 0-1.97-.25-2.8-.7l-1.46 1.46C8.97 19.54 10.43 20 12 20c4.42 0 8-3.58 8-8h3l-4-4zM6 12c0-3.31 2.69-6 6-6 1.01 0 1.97.25 2.8.7l1.46-1.46C15.03 4.46 13.57 4 12 4c-4.42 0-8 3.58-8 8H1l4 4 4-4H6z"
            />
          </svg>
        </div>
        <div
          className="BrowserControlBar__control"
          onClick={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="HomeIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
            />
          </svg>
        </div>
      </div>
      <div
        className="MuiFormControl-root MuiTextField-root BrowserControlBar_url-input css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
      >
        <div
          className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1ptx2yq-MuiInputBase-root-MuiInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
            disabled={false}
            id=":r0:"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
            value="https://www.github.com"
          />
        </div>
      </div>
      <div
        className="BrowserControlBar__bookmark-control"
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="StarBorderIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m22 9.24-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
          />
        </svg>
      </div>
    </div>
    <div
      className="Browser__webview-container"
    >
      <webview
        allowpopups="true"
        partition="persist:user-partition"
        src="https://www.github.com"
      />
    </div>
  </div>
  <div>
    <div
      className=""
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "cursor": "row-resize",
          "height": "10px",
          "left": "0px",
          "position": "absolute",
          "top": "-5px",
          "userSelect": "none",
          "width": "100%",
        }
      }
    />
    <div
      className=""
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "cursor": "col-resize",
          "height": "100%",
          "position": "absolute",
          "right": "-5px",
          "top": "0px",
          "userSelect": "none",
          "width": "10px",
        }
      }
    />
    <div
      className=""
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "bottom": "-5px",
          "cursor": "row-resize",
          "height": "10px",
          "left": "0px",
          "position": "absolute",
          "userSelect": "none",
          "width": "100%",
        }
      }
    />
    <div
      className=""
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "cursor": "col-resize",
          "height": "100%",
          "left": "-5px",
          "position": "absolute",
          "top": "0px",
          "userSelect": "none",
          "width": "10px",
        }
      }
    />
    <div
      className=""
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "cursor": "ne-resize",
          "height": "20px",
          "position": "absolute",
          "right": "-10px",
          "top": "-10px",
          "userSelect": "none",
          "width": "20px",
        }
      }
    />
    <div
      className=""
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "bottom": "-10px",
          "cursor": "se-resize",
          "height": "20px",
          "position": "absolute",
          "right": "-10px",
          "userSelect": "none",
          "width": "20px",
        }
      }
    />
    <div
      className=""
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "bottom": "-10px",
          "cursor": "sw-resize",
          "height": "20px",
          "left": "-10px",
          "position": "absolute",
          "userSelect": "none",
          "width": "20px",
        }
      }
    />
    <div
      className=""
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "cursor": "nw-resize",
          "height": "20px",
          "left": "-10px",
          "position": "absolute",
          "top": "-10px",
          "userSelect": "none",
          "width": "20px",
        }
      }
    />
  </div>
</div>
`;

exports[`Browser should toggle fullsize 1`] = `
"<div style=\\"position: absolute; user-select: auto; width: 800px; height: 600px; display: flex; top: 0px; left: 0px; cursor: auto; transform: translate(0px,0px); max-width: 9007199254740991px; max-height: 9007199254740991px; box-sizing: border-box; flex-shrink: 0; z-index: 2;\\" class=\\"Browser__draggable-container react-draggable\\" id=\\"Browser__123qsdf\\" data-testid=\\"browser-window\\" data-id=\\"123qsdf\\">
  <div class=\\"Browser__container\\" style=\\"opacity: 0;\\">
    <div class=\\"BrowserTopBar__container\\">
      <div class=\\"BrowserTopBar__title\\"></div>
      <div class=\\"BrowserTopBar__controls\\">
        <div class=\\"BrowserTopBar__control-button close-button\\" data-testid=\\"close-browser\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"CloseIcon\\">
            <path d=\\"M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z\\"></path>
          </svg></div>
        <div class=\\"BrowserTopBar__control-button\\" data-testid=\\"toggle-enlarge-browser\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"CropSquareIcon\\">
            <path d=\\"M18 4H6c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H6V6h12v12z\\"></path>
          </svg></div>
        <div class=\\"BrowserTopBar__control-button\\" data-testid=\\"minimize-browser\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"MinimizeIcon\\">
            <path d=\\"M6 19h12v2H6z\\"></path>
          </svg></div>
      </div>
    </div>
    <div class=\\"BrowserControlBar__container\\">
      <div class=\\"BrowserControlBar__controls\\">
        <div class=\\"BrowserControlBar__control\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"ArrowBackIcon\\">
            <path d=\\"M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z\\"></path>
          </svg></div>
        <div class=\\"BrowserControlBar__control\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"ArrowForwardIcon\\">
            <path d=\\"m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z\\"></path>
          </svg></div>
        <div class=\\"BrowserControlBar__control\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"CachedIcon\\">
            <path d=\\"m19 8-4 4h3c0 3.31-2.69 6-6 6-1.01 0-1.97-.25-2.8-.7l-1.46 1.46C8.97 19.54 10.43 20 12 20c4.42 0 8-3.58 8-8h3l-4-4zM6 12c0-3.31 2.69-6 6-6 1.01 0 1.97.25 2.8.7l1.46-1.46C15.03 4.46 13.57 4 12 4c-4.42 0-8 3.58-8 8H1l4 4 4-4H6z\\"></path>
          </svg></div>
        <div class=\\"BrowserControlBar__control\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"HomeIcon\\">
            <path d=\\"M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z\\"></path>
          </svg></div>
      </div>
      <div class=\\"MuiFormControl-root MuiTextField-root BrowserControlBar_url-input css-1u3bzj6-MuiFormControl-root-MuiTextField-root\\">
        <div class=\\"MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1ptx2yq-MuiInputBase-root-MuiInput-root\\"><input aria-invalid=\\"false\\" id=\\":r1:\\" type=\\"text\\" class=\\"MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input\\" value=\\"https://www.github.com\\"></div>
      </div>
      <div class=\\"BrowserControlBar__bookmark-control\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"StarBorderIcon\\">
          <path d=\\"m22 9.24-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z\\"></path>
        </svg></div>
    </div>
    <div class=\\"Browser__webview-container\\">
      <webview allowpopups=\\"true\\" src=\\"https://www.github.com\\" partition=\\"persist:user-partition\\"></webview>
    </div>
  </div>
  <div>
    <div class=\\"\\" style=\\"position: absolute; user-select: none; width: 100%; height: 10px; top: -5px; left: 0px; cursor: row-resize;\\"></div>
    <div class=\\"\\" style=\\"position: absolute; user-select: none; width: 10px; height: 100%; top: 0px; right: -5px; cursor: col-resize;\\"></div>
    <div class=\\"\\" style=\\"position: absolute; user-select: none; width: 100%; height: 10px; bottom: -5px; left: 0px; cursor: row-resize;\\"></div>
    <div class=\\"\\" style=\\"position: absolute; user-select: none; width: 10px; height: 100%; top: 0px; left: -5px; cursor: col-resize;\\"></div>
    <div class=\\"\\" style=\\"position: absolute; user-select: none; width: 20px; height: 20px; right: -10px; top: -10px; cursor: ne-resize;\\"></div>
    <div class=\\"\\" style=\\"position: absolute; user-select: none; width: 20px; height: 20px; right: -10px; bottom: -10px; cursor: se-resize;\\"></div>
    <div class=\\"\\" style=\\"position: absolute; user-select: none; width: 20px; height: 20px; left: -10px; bottom: -10px; cursor: sw-resize;\\"></div>
    <div class=\\"\\" style=\\"position: absolute; user-select: none; width: 20px; height: 20px; left: -10px; top: -10px; cursor: nw-resize;\\"></div>
  </div>
</div>"
`;
